src/main/java/org/eeichinger/servicevirtualisation/jdbc/WireMockMappingJsonRecorder.java

```diff
@@ -0,0 +1,132 @@
+package org.eeichinger.servicevirtualisation.jdbc;
+
+import lombok.Setter;
+import lombok.SneakyThrows;
+
+import java.io.FileWriter;
+import java.io.PrintWriter;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * Records called JDBC calls in the mapping json format that WireMock uses (e.g. expectedRecording.json)
+ * These are saved sequentially in /test/resources/newMappings/, e.g. 1.json, 2.json. This directory should exist.
+ * Workflow: these files can be renamed and move to the directory that WireMock is expecting (e.g. /test/resources/mappings)
+ */
+public class WireMockMappingJsonRecorder {
+
+    @Setter
+    private XmlTypeRegistry xmlTypeRegistry;
+
+    public WireMockMappingJsonRecorder() {
+        this(false);
+    }
+
+    public WireMockMappingJsonRecorder(boolean useExplicitTypes) {
+        if (useExplicitTypes) {
+            xmlTypeRegistry = new XmlTypeRegistry();
+        }
+    }
+
+    private int statementInTest = 0;
+
+    @SneakyThrows
+    protected String getSybaseXMLFromDBResult(ResultSet resultSet) {
```

Comment:
Fairly hacky, but just wanted something that works. Probably can make it a ton easier to read the files, but will leave that for a later exercise (if ever)